go-controller/pkg/ovn/address_set.go

```diff
@@ -25,34 +25,20 @@ type AddressSetFactory interface {
 	ForEachAddressSet(iteratorFn AddressSetIterFunc) error
 }
 
-type ovnAddressSetFactory struct {
-	sync.Mutex
-	sets map[string]*ovnAddressSet
-}
+type ovnAddressSetFactory struct{}
```

Comment:
OK now I see your intention here I think. The gress policy will hold the references and this is not meant to be a manager of all address sets. That's fine, but now you can just remove the factory. It has no attributes, so you can just call NewAddressSet from the lib.

Reply:
@trozet not if we want to fake out the AddressSet out we can't